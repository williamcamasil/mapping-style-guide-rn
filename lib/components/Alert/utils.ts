import { ViewStyle } from 'react-native';

import { Icons } from '../../assets';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type AlertVariantType =
  'success' |
  'info' |
  'warning' |
  'error';

export const getVariantColorStyle = (variant?: AlertVariantType): ColorPaletesType => {
  switch (variant) {
    case 'warning':
      return 'feedbackAlert500';
    case 'error':
      return 'feedbackError500';
    case 'info':
      return 'neutralGray600';
    case 'success':
    default:
      return 'feedbackSuccess500';
  }
};

export const getDefaultIcon = (variant?: AlertVariantType) => {
  switch (variant) {
    case 'success':
      return Icons.Default.Success;
    case 'info':
      return Icons.Default.Information;
    case 'warning':
    case 'error':
    default:
      return Icons.Default.Warning;
  }
};

export const getAlertStyle = (
  theme: AppThemeType,
  variant?: AlertVariantType,
): ViewStyle => {

  switch (variant) {
    case 'warning':
      return {
        backgroundColor: theme.colors.feedbackAlert100,
      };
    case 'error':
      return {
        backgroundColor: theme.colors.feedbackError100,
      };
    case 'info':
      return {
        backgroundColor: theme.colors.neutralGray100,
      };
    case 'success':
    default:
      return {
        backgroundColor: theme.colors.feedbackSuccess100,
      };
  }
};
