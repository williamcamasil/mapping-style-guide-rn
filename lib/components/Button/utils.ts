import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
