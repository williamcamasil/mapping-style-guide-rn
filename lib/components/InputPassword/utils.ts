import { StyleProp, ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export const INPUT_PASSWORD_HEIGHT = 64;

export const getCircleStatusStyle = (
  theme: AppThemeType,
  index: number,
  enteredNumbers: number,
  focused: boolean,
): ViewStyle | null => {
  if (!enteredNumbers && !focused) return null;

  // Circle focused
  if (index === enteredNumbers && focused) {
    return {
      backgroundColor: theme.colors.neutralGray200,
      borderColor: theme.colors.neutralGray600,
    };
  }

  // Circle filled
  if (index < enteredNumbers) {
    return {
      backgroundColor: theme.colors.primaryMain,
      borderWidth: theme.borders.width.thick,
      borderColor: theme.colors.neutralGray200,
    };
  }

  return null;
};

export const isCurrentEmptyTextFocused = (
  focused: boolean,
  enteredNumbers: number,
  index: number,
) => focused && enteredNumbers === index;

export const getTextPasswordColor = (
  index: number,
  enteredNumbers: number,
  focused: boolean,
  value?: string,
): ColorPaletesType => {
  if (value) return 'neutralGray600';
  if (isCurrentEmptyTextFocused(focused, enteredNumbers, index)) return 'neutralGray800';
  return 'neutralGray400';
};

export function getContentCirclesStyles(
  stretch: boolean | undefined,
  maxLength: number,
  theme: AppThemeType,
): StyleProp<ViewStyle> {

  if (stretch) {
    return {
      width: undefined,
      flex: 1,
      justifyContent: 'space-between',
    };
  }
  const circleSize = theme.spacings.sSmall;
  const circleSpacing = theme.spacings.sXS;
  const quantitySpacings = maxLength - 1;

  return {
    width: maxLength * circleSize + quantitySpacings * circleSpacing,
    flex: 0,
    justifyContent: 'flex-start',
  };
}
