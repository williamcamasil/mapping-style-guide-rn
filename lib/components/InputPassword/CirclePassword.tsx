import React from 'react';
import { View } from 'react-native';

import { useViewStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import { getCircleStatusStyle } from './utils';

type CirclePasswordPropType = {
  theme: AppThemeType;
  index: number;
  value: number;
  focused: boolean;
};

const CirclePassword: React.FC<CirclePasswordPropType> = ({
  theme, index, value, focused,
}) => {

  const circleViewStyle = useViewStyles(() => [
    {
      width: theme.spacings.sSmall,
      height: theme.spacings.sSmall,
      borderRadius: theme.spacings.sSmall / 2,
      backgroundColor: theme.colors.neutralWhite,
      borderColor: theme.colors.neutralGray400,
      borderWidth: theme.borders.width.thin,
    },
    getCircleStatusStyle(theme, index, value, focused),
  ], [index, focused, theme, value]);

  return (
    <View
      testID={`circle-touchable-${index}`}
      pointerEvents="none"
      style={circleViewStyle}
    />
  );
};

export default withTheme(CirclePassword);
