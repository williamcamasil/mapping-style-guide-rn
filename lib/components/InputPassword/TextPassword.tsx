import React, { useCallback } from 'react';
import { StyleSheet, View } from 'react-native';

import { useTextStyles } from '../../hooks';
import { AppThemeType, withTheme } from '../../theme';
import Text from '../Text';
import { getTextPasswordColor, isCurrentEmptyTextFocused } from './utils';

type TextPasswordPropType = {
  theme: AppThemeType;
  index: number;
  value: string;
  focused: boolean;
};

const styles = StyleSheet.create({
  containerText: {
    width: 15,
  },
});

const TextPassword: React.FC<TextPasswordPropType> = ({
  theme, index, value, focused,
}) => {

  const char = value[index];
  const enteredNumbers = value.length;

  const textVisibleStyles = useTextStyles(() => [
    {
      paddingLeft: theme.spacings.sQuark,
    },
    styles.containerText,
  ], [theme]);

  const getTextPasswordIndexContent = useCallback(() => {
    if (char) return char;
    if (isCurrentEmptyTextFocused(focused, enteredNumbers, index)) return '|';
    return '0';
  }, [char, enteredNumbers, focused, index]);

  return (
    <View
      testID={`text-password-touchable-${index}`}
      pointerEvents="none"
      style={textVisibleStyles}
    >
      <Text variant="subtitle" weight="semiBold" color={getTextPasswordColor(index, enteredNumbers, focused, char)}>
        {getTextPasswordIndexContent()}
      </Text>
    </View>
  );
};

export default withTheme(TextPassword);
