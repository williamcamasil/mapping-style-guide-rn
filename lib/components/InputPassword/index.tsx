import React, {
  useCallback, useMemo, useState,
} from 'react';
import {
  NativeSyntheticEvent,
  StyleSheet,
  TextInput,
  TextInputFocusEventData,
  TextInputProps,
  View,
} from 'react-native';

import { Icons } from '../../assets';
import { useTextStyles, useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { wrapField } from '../Form';
import Spacer from '../Spacer';
import CirclePassword from './CirclePassword';
import TextPassword from './TextPassword';
import {
  INPUT_PASSWORD_HEIGHT, getContentCirclesStyles,
} from './utils';

export type InputPasswordPropsType = Omit<TextInputProps, 'placeholder' | 'onChange' | 'onChangeText' | 'editable' | 'pointerEvents' | 'maxLength'> & {
  theme: AppThemeType;
  value: string | null;
  onChange?: TextInputProps['onChangeText'];
  keyboardType?: TextInputProps['keyboardType'];
  maxLength: number;
  innerRef?: React.LegacyRef<TextInput>;
  stretch?: boolean;
};

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
  },
  containerCircles: {
    left: 0,
    right: 0,
    position: 'absolute',
    flexDirection: 'row',
    alignItems: 'center',
  },
  contentCircle: {
    flexDirection: 'row',
  },
});

const InputPassword = ({
  theme,
  value,
  onChange,
  testID,
  innerRef,
  onFocus,
  onBlur,
  stretch,
  maxLength,
  ...others
}: InputPasswordPropsType): React.ReactElement => {
  const [focused, setFocused] = useState(false);
  const [visiblePassword, setVisiblePassword] = useState(false);

  const inputStyles = useTextStyles(() => [
    {
      borderColor: theme.colors.neutralWhite,
      borderRadius: theme.borders.radius.medium,
      height: INPUT_PASSWORD_HEIGHT,
      borderWidth: theme.borders.width.thin,
      opacity: theme.opacities.transparent,
    },
  ], [theme]);

  const iconStyles = useViewStyles(() => [
    {
      marginLeft: theme.spacings.sMedium,
    },
  ], [theme.spacings.sMedium]);

  const contentCirclesStyles = useViewStyles(() => [
    {
      flexDirection: 'row',
    },
    getContentCirclesStyles(stretch, maxLength, theme),
  ], [maxLength, stretch, theme]);

  const handleFocus = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(true);
      onFocus?.(e);
    },
    [onFocus],
  );

  const handleBlur = useCallback(
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => {
      setFocused(false);
      onBlur?.(e);
    },
    [onBlur],
  );

  const listPositions = useMemo(() => [...Array(maxLength)].map((_, index) => ({ index })), [maxLength]);

  const renderCirclesPassword = useCallback(() => listPositions.map(item => {
    const lastIndex = listPositions.length - 1;
    const isLastItem = item.index === lastIndex;
    return (
      <View key={item.index} style={styles.contentCircle} pointerEvents="box-none">
        <CirclePassword
          key={item.index}
          theme={theme}
          index={item.index}
          value={value.length}
          focused={focused}
        />
        {
          isLastItem ? null : <Spacer size={theme.spacings.sXS} />
        }
      </View>
    );
  }), [focused, listPositions, theme, value.length]);

  const renderTextsPassword = useCallback(() => listPositions.map(item => {
    const lastIndex = listPositions.length - 1;
    const isLastItem = item.index === lastIndex;
    return (
      <View key={item.index} style={styles.contentCircle} pointerEvents="box-none">
        <TextPassword
          key={item.index}
          theme={theme}
          index={item.index}
          value={value}
          focused={focused}
        />
        {
          isLastItem ? null : <Spacer size={theme.spacings.sSmall} />
        }
      </View>

    );
  }), [focused, listPositions, theme, value]);

  const inputSelection = useMemo(() => ({
    /*
     * ForÃ§a o cursor para ficar sempre no FINAL do input.
     */
    start: value.length,
    end: value.length,
  }), [value.length]);

  const changeVisiblePassword = useCallback(() => {
    setVisiblePassword(prevState => !prevState);
  }, []);

  const renderVisiblePasswordIcon = (): JSX.Element => {
    const Icon = visiblePassword ? Icons.Default.EyeOff : Icons.Default.EyeOn;
    const testId = visiblePassword ? 'btn-icon-hide-password' : 'btn-icon-show-password';
    return (
      <Icon
        testID={testId}
        style={iconStyles}
        onPress={changeVisiblePassword}
        color={theme.colors.neutralGray400}
      />
    );
  };

  return (
    <View testID={testID} style={styles.container}>
      <TextInput
        {...others}
        ref={innerRef}
        testID="rn-text-input"
        caretHidden
        secureTextEntry
        value={value}
        onChangeText={onChange}
        onFocus={handleFocus}
        onBlur={handleBlur}
        style={inputStyles}
        selection={inputSelection}
      />
      <View style={styles.containerCircles} pointerEvents="box-none">
        <View style={contentCirclesStyles} pointerEvents="box-none">
          {visiblePassword ? renderTextsPassword() : renderCirclesPassword()}
        </View>
        {renderVisiblePasswordIcon()}
      </View>
    </View>
  );
};

InputPassword.defaultProps = {
  keyboardType: 'numeric' as TextInputProps['keyboardType'],
};

InputPassword.Field = withTheme(wrapField<InputPasswordPropsType>((props, fieldProps) => {
  const {
    value, onChange, onBlur, onFocus,
  } = fieldProps;

  return (
    <InputPassword
      {...props}
      value={value || ''}
      onChange={onChange}
      onBlur={onBlur}
      onFocus={onFocus}
    />
  );
}));

export default withTheme(InputPassword);
