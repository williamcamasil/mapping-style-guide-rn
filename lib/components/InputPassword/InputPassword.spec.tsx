import React from 'react';

import { render, fireEvent } from '@testing-library/react-native';
import { createForm } from 'final-form';

import InputPassword from '.';
import { DefaultTheme } from '../../theme';
import Form from '../Form';
import { getCircleStatusStyle, getTextPasswordColor, isCurrentEmptyTextFocused } from './utils';

describe('InputPassword', () => {
  it('InputPassword snapshot', () => {
    const result = render(
      <InputPassword
        maxLength={6}
        value=""
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should render InputPassword with the property stretch', () => {
    const result = render(
      <InputPassword
        maxLength={6}
        value=""
        stretch
      />,
    ).toJSON();

    expect(result).toMatchSnapshot();
  });

  it('should onFocus on inputPassword', () => {
    const result = render(
      <InputPassword
        maxLength={6}
        value=""
      />,
    );

    const input = result.getByTestId('rn-text-input');
    fireEvent(input, 'focus');

    expect(result.toJSON).toMatchSnapshot();
  });

  it('should onChange on inputPassword', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputPassword
        maxLength={6}
        onChange={onChangeMock}
        value=""
      />,
    );

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', '123');

    expect(onChangeMock).toBeCalledWith('123');
  });

  it('should onClick on inputPassword circle', () => {
    const result = render(
      <InputPassword
        maxLength={6}
        value=""
      />,
    );

    const circleTouchable = result.getByTestId('circle-touchable-1');
    fireEvent.press(circleTouchable);

    expect(result.toJSON).toMatchSnapshot();
  });

  it('should on click to show password', () => {
    const result = render(
      <InputPassword
        maxLength={6}
        value=""
      />,
    );

    const iconShowPassword = result.getByTestId('btn-icon-show-password');

    fireEvent.press(iconShowPassword);

    expect(result.getByTestId('btn-icon-hide-password')).toBeDefined();
    expect(result.toJSON).toMatchSnapshot();
  });

  it('should change text with visible password', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputPassword
        maxLength={6}
        onChange={onChangeMock}
        value=""
      />,
    );

    const iconShowPassword = result.getByTestId('btn-icon-show-password');
    fireEvent.press(iconShowPassword);

    fireEvent(result.getByTestId('rn-text-input'), 'changeText', '123');

    expect(onChangeMock).toBeCalledWith('123');
    expect(result.toJSON).toMatchSnapshot();
  });

  it('should change text focused with visible password', async () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputPassword
        maxLength={6}
        onChange={onChangeMock}
        value="123"
      />,
    );

    const iconShowPassword = result.getByTestId('btn-icon-show-password');
    fireEvent.press(iconShowPassword);

    const rnTextInput = result.getByTestId('rn-text-input');

    fireEvent(rnTextInput, 'focus');
    fireEvent(rnTextInput, 'changeText', '123');

    const textView = await result.findByText('|');

    expect(textView).toBeDefined();
    expect(result.toJSON).toMatchSnapshot();
  });

  it('should change text focused with visible password and blur', () => {
    const onChangeMock = jest.fn();
    const result = render(
      <InputPassword
        maxLength={6}
        onChange={onChangeMock}
        value="123"
      />,
    );

    const iconShowPassword = result.getByTestId('btn-icon-show-password');
    fireEvent.press(iconShowPassword);

    const rnTextInput = result.getByTestId('rn-text-input');

    fireEvent(rnTextInput, 'focus');
    fireEvent(rnTextInput, 'changeText', '123');
    fireEvent(rnTextInput, 'blur');

    const textView = result.queryByText('|');

    expect(textView).toBeNull();
    expect(result.toJSON).toMatchSnapshot();
  });
});

describe('InputPassword Utils', () => {
  it('should getCircleStatusStyle', () => {
    const result = getCircleStatusStyle(DefaultTheme, 1, 6, false);

    expect(result?.backgroundColor).toEqual(DefaultTheme.colors.primaryMain);
    expect(result?.borderWidth).toEqual(DefaultTheme.borders.width.thick);
    expect(result?.borderColor).toEqual(DefaultTheme.colors.neutralGray200);
  });

  it('should getCircleStatusStyle focused', () => {
    const result = getCircleStatusStyle(DefaultTheme, 5, 4, true);

    expect(result).toBeDefined();

    expect(result?.backgroundColor).toBeUndefined();
    expect(result?.borderColor).toBeUndefined();
  });

  it('should call isCurrentEmptyTextFocused', () => {
    const resultFocused = isCurrentEmptyTextFocused(true, 6, 6);
    expect(resultFocused).toBeTruthy();

    const resultUnfocused = isCurrentEmptyTextFocused(false, 6, 6);
    expect(resultUnfocused).toBeFalsy();

    const resultFocusedMoreEntryNumbers = isCurrentEmptyTextFocused(false, 6, 1);
    expect(resultFocusedMoreEntryNumbers).toBeFalsy();
  });

  it('should getTextPasswordColor', () => {
    const resultValueDefined = getTextPasswordColor(1, 1, false, '1');
    expect(resultValueDefined).toEqual('neutralGray600');

    const resultIsCurrentEmptyTextFocused = getTextPasswordColor(1, 1, true);
    expect(resultIsCurrentEmptyTextFocused).toEqual('neutralGray800');

    const elseResult = getTextPasswordColor(1, 1, false);
    expect(elseResult).toEqual('neutralGray400');
  });
});

describe('InputPassword.Field', () => {
  it('valid', async () => {
    const form = createForm({
      onSubmit: jest.fn(),
    });

    const component = render(
      <Form
        onSubmit={jest.fn()}
        form={form}
      >
        {() => (
          <InputPassword.Field
            name="inputPassword"
            maxLength={6}
            required
            validate={value => (value?.length === 6 ? undefined : 'Erro')}
          />
        )}
      </Form>,
    );

    expect(form.getState().valid).toBeFalsy();

    const inputSelect = component.getByTestId('rn-text-input');
    fireEvent(inputSelect, 'changeText', '123456');

    expect(form.getState().valid).toBeTruthy();
  });
});
