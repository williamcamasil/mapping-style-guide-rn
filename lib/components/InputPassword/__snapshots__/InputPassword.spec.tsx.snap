// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputPassword InputPassword snapshot 1`] = `
<View
  style={
    {
      "justifyContent": "center",
    }
  }
>
  <TextInput
    caretHidden={true}
    keyboardType="numeric"
    onBlur={[Function]}
    onFocus={[Function]}
    secureTextEntry={true}
    selection={
      {
        "end": 0,
        "start": 0,
      }
    }
    style={
      {
        "borderColor": "#FFFFFF",
        "borderRadius": 12,
        "borderWidth": 1,
        "height": 64,
        "opacity": 0,
      }
    }
    testID="rn-text-input"
    value=""
  />
  <View
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        {
          "flex": 0,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": 200,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-0"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-1"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-2"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-3"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-4"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-5"
        />
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      onPress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      responsible={true}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginLeft": 24,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="btn-icon-show-password"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M2.13 12.297a1.097 1.097 0 0 1 0-1.04C4.234 7.37 8.118 4 12 4c3.883 0 7.767 3.37 9.87 7.259.174.323.174.715 0 1.039-2.103 3.888-5.987 7.258-9.87 7.258-3.883 0-7.767-3.37-9.87-7.26Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
        <RNSVGPath
          d="M14.357 9.42a3.333 3.333 0 1 1-4.714 4.715 3.333 3.333 0 0 1 4.714-4.714Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`InputPassword should change text focused with visible password 1`] = `[Function]`;

exports[`InputPassword should change text focused with visible password and blur 1`] = `[Function]`;

exports[`InputPassword should change text with visible password 1`] = `[Function]`;

exports[`InputPassword should on click to show password 1`] = `[Function]`;

exports[`InputPassword should onClick on inputPassword circle 1`] = `[Function]`;

exports[`InputPassword should onFocus on inputPassword 1`] = `[Function]`;

exports[`InputPassword should render InputPassword with the property stretch 1`] = `
<View
  style={
    {
      "justifyContent": "center",
    }
  }
>
  <TextInput
    caretHidden={true}
    keyboardType="numeric"
    onBlur={[Function]}
    onFocus={[Function]}
    secureTextEntry={true}
    selection={
      {
        "end": 0,
        "start": 0,
      }
    }
    style={
      {
        "borderColor": "#FFFFFF",
        "borderRadius": 12,
        "borderWidth": 1,
        "height": 64,
        "opacity": 0,
      }
    }
    testID="rn-text-input"
    value=""
  />
  <View
    pointerEvents="box-none"
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-0"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-1"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-2"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-3"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-4"
        />
        <Text
          style={
            {
              "height": 16,
              "width": 16,
            }
          }
        />
      </View>
      <View
        pointerEvents="box-none"
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#C2C5C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 20,
              "width": 20,
            }
          }
          testID="circle-touchable-5"
        />
      </View>
    </View>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height={24}
      meetOrSlice={0}
      minX={0}
      minY={0}
      onPress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      responsible={true}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "marginLeft": 24,
          },
          {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      testID="btn-icon-show-password"
      vbHeight={24}
      vbWidth={24}
      width={24}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M2.13 12.297a1.097 1.097 0 0 1 0-1.04C4.234 7.37 8.118 4 12 4c3.883 0 7.767 3.37 9.87 7.259.174.323.174.715 0 1.039-2.103 3.888-5.987 7.258-9.87 7.258-3.883 0-7.767-3.37-9.87-7.26Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
        <RNSVGPath
          d="M14.357 9.42a3.333 3.333 0 1 1-4.714 4.715 3.333 3.333 0 0 1 4.714-4.714Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4290954697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="1.8"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
